inspirehep/modules/workflows/tasks/upload.py

```diff
@@ -97,7 +99,10 @@ def store_record(obj, eng):
             record.commit()
             obj.save()
     else:
-        store_record_inspirehep_api(obj, eng, is_update, is_authors)
+        try:
+            store_record_inspirehep_api(obj, eng, is_update, is_authors)
+        except WorkflowsError:
+            restart_workflow(obj)
```

Comment:
hmm infinite loop? we should increase the restart counter first

Reply:
afair it does it automatically, will check

Reply:
if I understand correctly the `restart_workflow` is restarting from step `0` by default so it goes through all the steps including `increase_restart_count` [in pre-processing](https://github.com/inspirehep/inspire-next/blob/a83bae4dc8be2f091d25b6b16c3576b07e8744ba/inspirehep/modules/workflows/workflows/article.py#L587), so imo it's not needed to increase restart count here.